File: app/src/main/java/br/pedroso/upcomingmovies/network/services/TheMovieDbService.java
package br.pedroso.upcomingmovies.network.services;

import java.util.List;

import br.pedroso.upcomingmovies.network.entities.RetrofitMovieEntity;
import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

public interface TheMovieDbService {
    String BASE_URL = "https://api.themoviedb.org/3/";

    @GET("movie/upcoming")
    Observable<List<RetrofitMovieEntity>> listUpcomingMovies();

    @GET("movie/{movieId}")
    Observable<RetrofitMovieEntity> getMovieDetails(@Path("movieId") Integer movieId);


    @GET("movie/{movieId}/similar")
    Observable<List<RetrofitMovieEntity>> listSimilarMovies(@Path("movieId") Integer movieId);
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/network/services/TheMovieDbService.kt
package br.pedroso.upcomingmovies.network.services

import br.pedroso.upcomingmovies.network.entities.NetworkMovie
import br.pedroso.upcomingmovies.network.entities.NetworkMoviesResponse

interface TheMovieDbService {
    suspend fun listUpcomingMovies(): NetworkMoviesResponse

    suspend fun getMovieDetails(movieId: Int): NetworkMovie

    suspend fun listSimilarMovies(movieId: Int): NetworkMoviesResponse

    companion object {
        const val BASE_URL: String = "https://api.themoviedb.org/3/"
    }
}
